src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -849,6 +849,26 @@ private function setReturnType(string $types, string $class, string $method, str
         $docTypes = [];
 
         foreach ($typesMap as $n => $t) {
+            if (str_contains($n, '::')) {
+                [$definingClass, $constantName] = explode('::', $n, 2);
+                $definingClass = match ($definingClass) {
+                    'self', 'static', 'parent' => $class,
+                    default => $definingClass,
+                };
+
+                if (!\defined($definingClass.'::'.$definingClass)) {
```

Comment:
We ignore them for now. The changes from this PR require that you use the full constant name. We could ship support for wildcards in a follow-up PR (I would like to ship this one as soon as possible to make the job green again and supporting wildcards requires a bit more work).

Reply:
I'm fine with ignoring them (we were already ignoring those cases before, see the comment below). But I'm wondering whether this will be ignored cleaning here.

Reply:
Yes, returning here (which means not reaching the line where the `$returnTypes` property is written for the current method means that no patching happens afterwards. The same is done for example further down for the `resource` type.